public/modules/custom/helfi_rekry_content/src/Plugin/Deriver/HelbitMigrationDeriver.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Drupal\helfi_rekry_content\Plugin\Deriver;
+
+use Drupal\Component\Plugin\Derivative\DeriverBase;
+use Drupal\Core\Plugin\Discovery\ContainerDeriverInterface;
+use Drupal\Core\Url;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Deriver for Helbit migrations.
+ */
+class HelbitMigrationDeriver extends DeriverBase implements ContainerDeriverInterface {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, $base_plugin_id) {
+    return new static();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDerivativeDefinitions($base_plugin_definition) {
+    $derivatives = [
+      'all',
+      'changed',
+      'all_sv',
+      'changed_sv',
+      'all_en',
+      'changed_en',
+    ];
+
+    foreach ($derivatives as $key) {
+      $derivative = $this->getDerivativeValues($base_plugin_definition, $key);
+      $this->derivatives[$key] = $derivative;
+    }
+
+    return $this->derivatives;
+  }
+
+  /**
+   * Creates a derivative definition for each available language.
+   *
+   * @param array $base_plugin_definition
+   *   Base migration definitions.
+   * @param string $key
+   *   Key for derivative.
+   *
+   * @return array
+   *   Modified plugin definition for derivative.
+   */
+  protected function getDerivativeValues(array $base_plugin_definition, string $key): array {
+    $urlOptions = [
+      'query' => [
+        'client' => getenv('HELBIT_CLIENT_ID'),
```

Comment:
This should probably be converted to a proper configuration that can be overridden in `settings.php`.

Like `client => $this->config->get('helbit_client_id');`

Then in all.settings.php:
```php
$config['helfi_rekry_content.settings']['helbit_client_id'] = getenv('HELBIT_CLIENT_ID');`
```